core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -73,6 +76,10 @@
   // A flag telling whether metrics have been reported yet.
   // Using this prevents us from initializing {@link #SHOULD_REPORT_METRICS} more than once
   private static boolean sReported = false;
+  // The source of the metrics in this metrics system.
+  // It can be set through property keys based on process types.
+  // Local hostname will be used if no related property key founds.
+  private static String sSourceName;
```

Comment:
Why does this need to be a static variable? It seems like we are using it as a local variable?

Reply:
All the methods calling it are static methods.

Reply:
All methods in the `MetricsSystem` class are static methods, MetricsSystem is initialized per-jvm and does not allow explicit initialization.

Reply:
I mean a local variable in the method, not an instance variable

Reply:
Oh, I understand, this is used for memory the source value since this method `getMetricNameWithUniqueId` is called really often.

I added the null check